apps/ejabberd/src/mod_mam_utils.erl

```diff
@@ -346,6 +346,25 @@ is_valid_message_children(_,     false, false, false) -> true;
 %% See mam_SUITE:offline_message for a test case
 is_valid_message_children(_,      _,    _,     _    ) -> false.
 
+is_chat_marker(Packet) ->
+    case mod_mam_params:archive_chat_markers() of
+        true ->
+            do_check_markers(Packet);
+        false ->
+            false
+    end.
+
+do_check_markers(Packet) ->
+    MarkerTags = [<<"received">>, <<"displayed">>, <<"acknowledged">>],
+    HasSubtag = fun(Tag) ->
+                    case xml:get_subtag(Packet, Tag) of
```

Comment:
Master is still not using `exml` with new selector. It's a post-beta2 task to change it and also update the as many parts of the code as possible to use it.